### PR TITLE
chore(test): karma-webpack, test both CommonJS & ES modules

### LINKED ISSUE #543: Packaged artifacts not tested
The two sets of build artifacts that this project provides — CommonJS and ES modules — are not being tested currently. Instead, karma-typescript builds its own bundle (likely in-memory) and that what is being tested.

https://github.com/snabbdom/snabbdom/blob/d4a8640132ceb135c21eeda84e2c0b58828eff5a/karma.conf.js#L17-L64

### PR DESCRIPTION
**Pull request by mightyiam**
Fixes #543.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -17,19 +17,28 @@ const browsers = ci
 module.exports = function (config) {
   config.set({
     basePath: '.',
-    frameworks: ['mocha', 'karma-typescript'],
+    frameworks: ['mocha'],
     // list of files / patterns to load in the browser
-    files: [{ pattern: 'src/**/*' }],
+    files: [
+      { pattern: 'test/**/*.js' },
+      { pattern: 'es/test/**/*.js' }
```
Comment (mightyiam) [inline karma.conf.js]: Tests are run twice — once for CommonJS and once for ES modules.

File: karma.conf.js [hunk 2]
```diff
@@ -40,20 +49,7 @@ module.exports = function (config) {
...
-      },
-      compilerOptions: {
-        module: 'commonjs'
-      },
-      tsconfig: './tsconfig.json',
-      include: {
-        mode: 'merge',
-        values: ['src/test/**/*'],
-      },
-    },
```
Comment (mightyiam) [inline karma.conf.js]: And it's simpler.

**Comment by kuraga** [comment]
@mightyiam thanks very much!

1.
```
    "compile-es": "tsc --outDir es --module es6",
    "compile-commonjs": "tsc --outDir . --module commonjs",
```
move these artifacts in a subfolder, maybe?

2. We now use Webpack (for browser testing). Get rid of Browserify, maybe?

**Comment by mightyiam** [comment]
@kuraga you're welcome :bouquet: 

Yes, I'd love to move at least the CommonJS build into a subfolder. Just to be clear — this suggestion is not related to this PR, right?

Browserify is still used to bundle some examples and the UMD bundles. Both of these are on my radar.

**Comment by kuraga** [comment]
@mightyiam is not related. Just too much to open two issues for this :smile: And new code uses compilation.

> Browserify is still used to bundle some examples and the UMD bundles. Both of these are on my radar.

:+1:

**Comment by mightyiam** [comment]
@kuraga what do you mean by "new code uses compilation", please? What new code and what compilation, please?

**Pull request merged**